### PR TITLE
Dataset-JSON 1.1

### PR DESCRIPTION
**Pull request by TeMeta**
Dataset-JSON 1.1 major version

* Overhauled schema based on community feedback
* Switched from TS to LinkML for more varied modelling support
* Added tooling for conversion from 1.0 to 1.1
* Added Jupyter Notebook for easier interaction
* Updated all examples

### CODE REVIEW

File: convert/convert_1_0_to_1_1.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        "fileOID": old.get("fileOID", ""),
+        "label": old.get("fileOID", ""),
+        "metaDataRef": old.get("metaDataRef", ""),
+        "metaDataVersionOID": old.get("metaDataVersionOID", ""),
+        "name": old.get("fileOID", ""), 
+        "originator": old.get("originator", ""),
+        "records": 0,
+        "sourceSystem": old.get("sourceSystem", ""),
+        "sourceSystemVersion": old.get("sourceSystemVersion", ""),
+        "studyOID": old.get("studyOID", ""),
```
Comment (glow-mdsol) [inline convert/convert_1_0_to_1_1.py:19]: `studyOID` is in `clinicalData` in V1.0 I think

File: convert/convert_1_0_to_1_1.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+def convert_1_0_to_1_1(old):
+    new = {
+        "asOfDateTime": old.get("asOfDateTime", ""),
+        "columns": [],
+        "creationDateTime": old.get("creationDateTime", ""),
+        "datasetJSONVersion": old.get("datasetJSONVersion", ""),
+        "fileOID": old.get("fileOID", ""),
+        "label": old.get("fileOID", ""),
+        "metaDataRef": old.get("metaDataRef", ""),
```
Comment (glow-mdsol) [inline convert/convert_1_0_to_1_1.py:12]: `metaDataRef` is in `clinicalData` in V1.0 I think

File: convert/convert_1_0_to_1_1.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        "metaDataVersionOID": old.get("metaDataVersionOID", ""),
```
Comment (glow-mdsol) [inline convert/convert_1_0_to_1_1.py:13]: `metaDataVersionOID` is in `clinicalData` in V1.0 I think

File: convert/convert_1_0_to_1_1.py (hunk 1 above)
Comment (TeMeta) [reply convert/convert_1_0_to_1_1.py]: Good spots!

**Comment by jehugaleahsa** [comment]
Should the dataset-json version be updated to 1.1.0 in these files?

**Comment by TeMeta** [comment]
Need to set `datasetJSONVersion` to "1.1.0"

**Comment by lexjansen** [comment]
I think it would be good to add the SEND examples from https://github.com/cdisc-org/DataExchange-DatasetJson/pull/21 as well. We want to make sure SEND is represented.

**Comment by lexjansen** [comment]
The examples need some refinement.
Example adam\adlbcpv.json:

`
            "dataType": "integer",
            "displayFormat": "DATE9.",
            "itemOID": "IT.ADLBCPV.TRTSDT",
            "keySequence": 0,
            "label": "Date of First Exposure to Treatment",
            "length": 8,
            "name": "TRTSDT",
            "targetDataType": "integer"
        },
`

In this case dataType should be "date".
Also. the keySequence should not be 0 when the item is not part of key variable in the dataset structure. I would prefer not to have the keySequence key then.

Also, not wrong, but:
- if targetDataType is the same as dataType, do we need to specify it? 
- if displayFormat is null, do we even need to have the key.
- I would prefer the JSON files not be pretty printed, so that people get used to this. You really don't want pretty-printing in a submission. Common tools can format with indentaton on-the-fly.

**Comment by TeMeta** [comment]
> I think it would be good to add the SEND examples from #21 as well. We want to make sure SEND is represented.

Good point. Can we do it after this PR has been corrected and approved, then update the SEND examples the same way upon merging #21 to the version 1.1?

**Comment by lexjansen** [comment]
Looks good!
One more use case to illustrate targetDataType. In the SDTM datasets there are decimal data types.
They also need targetDataType.

Example:
```
        {
            "dataType": "decimal",
            "itemOID": "IT.VS.VSSTRESN",
            "label": "Numeric Result/Finding in Standard Units",
            "length": 8,
            "name": "VSSTRESN"
        },
```
I think this would be:
```
        {
            "dataType": "string",
            "itemOID": "IT.VS.VSSTRESN",
            "label": "Numeric Result/Finding in Standard Units",
            "length": 8,
            "name": "VSSTRESN"
            "targetDataType": "decimal",
        },
```

**Comment by lexjansen** [comment]
Great that you added the order!
I think you missed a few:
```
    "metaDataRef": "https://metadata.location.org/CDISCPILOT01/define.xml",
    "metaDataVersionOID": "MDV.MSGv2.0.SDTMIG.3.3.SDTM.1.7",
    "studyOID": "cdisc.com/CDISCPILOT01",
    "itemGroupOID": "IG.VS"

```

**Comment by TeMeta** [comment]
> Great that you added the order! I think you missed a few:
> 
> ```
>     "metaDataRef": "https://metadata.location.org/CDISCPILOT01/define.xml",
>     "metaDataVersionOID": "MDV.MSGv2.0.SDTMIG.3.3.SDTM.1.7",
>     "studyOID": "cdisc.com/CDISCPILOT01",
>     "itemGroupOID": "IG.VS"
> ```

Hi Lex, this has been fixed now. I used `vs.json` from 1.0 as the reference for ordering elements
Thanks for the thorough review!

**Review by lexjansen — APPROVED** [review]

**Review by glow-mdsol — APPROVED** [review]
Looks great!

**Review by swhume — APPROVED** [review]
Overall, looks good. I believe that decimal is likely a targetDataType instead of a dataType, but I'll approve and we can update that in the next iteration if the team agrees. We still need to decide about date and time, so there may well be other data type related changes.

**Comment by lexjansen** [comment]
@swhume It is:
```
        {
            "itemOID": "IT.VS.VSSTRESN",
            "name": "VSSTRESN",
            "label": "Numeric Result/Finding in Standard Units",
            "dataType": "string",
            "targetDataType": "decimal",
            "length": 8
        },

```

**Pull request merged**